File: ios/ExpoCheckInstalledAppsModule.swift
import ExpoModulesCore
import UIKit

public class ExpoCheckInstalledAppsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoCheckInstalledApps")

    AsyncFunction("checkAppsInstalled") { (packageNames: [String]) -> [String: Bool] in
      var result: [String: Bool] = [:]

      for packageName in packageNames {
        let scheme = packageName.contains("://") ? packageName : "\(packageName)://"
        if let url = URL(string: scheme) {
          result[packageName] = UIApplication.shared.canOpenURL(url)
        } else {
          result[packageName] = false
        }
      }

      return result
    }.runOnQueue(.main)
  }
}

File: android/src/main/java/expo/modules/checkinstalledapps/ExpoCheckInstalledAppsModule.kt
package expo.modules.checkinstalledapps

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import android.content.pm.PackageManager

class ExpoCheckInstalledAppsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoCheckInstalledApps")

    AsyncFunction("checkAppsInstalled") { packageNames: List<String> ->
      val pm: PackageManager = requireNotNull(appContext.reactContext).packageManager
      val result = mutableMapOf<String, Boolean>()

      for (packageName in packageNames) {
        result[packageName] = try {
          pm.getPackageInfo(packageName, 0)
          true
        } catch (e: PackageManager.NameNotFoundException) {
          false
        }
      }

      result
    }
  }
}
